### PR TITLE
Fixed `escape_windows_cmd_string` not handling ^ correctly

### PR DESCRIPTION
**Pull request by melund**
@santagada Can you try "source-cmd" that failed for you with this fix?

### CODE REVIEW

File: xonsh/tools.py [hunk 1]
```diff
@@ -890,9 +890,8 @@ def escape_windows_cmd_string(s):
     The escaping is based on details here and empirical testing:
     http://www.robvanderwoude.com/escapechars.php
     """
-    for c in '()%!^<>&|"':
+    for c in '^()%!<>&|"':
         s = s.replace(c, '^' + c)
-    s = s.replace('/?', '/.')
```
Comment (melund) [inline xonsh/tools.py]: @rbrewer123 I removed this from the `escape_windows_cmd_string()`. I think it is something you once added, but I can't see what the purpose of it is.

**Comment by santagada** [comment]
I should be able to test this...

**Comment by santagada** [comment]
It works perfectly, thanks!

**Comment by laerus** [comment]
merging this then! can't test it myself though, thank you guys!

**Comment by melund** [comment]
Thanks a lot @laerus

**Pull request merged**